### ISSUE TITLE
skip light immune Nilhathak

### ISSUE THREAD

**Issue by drderpman** [kolton/d2bot-with-kolbot#1276]
Hi, I've noticed that my light sorc sometimes dies at Nilhathak because he is light immune sometimes and she dies waiting for my merc to kill him. Is there a way to have her skip Nilhathak only when he is light immune. She can handle other light immunes so I dont want to skip them, just Nihlathank when he is light immune. Thanks

**Comment by aim2kill** [comment]
Hi, a question are using tele.stomp, do you infinity?

**Comment by expl0** [comment]
Insert the following:

```
var unit = getUnit(1, 526); // Nihlathak
if (unit && !Attack.checkResist(unit, "lightning")) {
Attack.kill(526); // Nihlathak
}
```

**Comment by drderpman** [comment]
> Hi, a question are using tele.stomp, do you infinity?

Yes infinity on and tele.stomp on

**Comment by aim2kill** [comment]
Do you have priority to kill boss over minions? Sometime that help, aleast my sorc kill him fast like that

**Comment by drderpman** [comment]
> Do you have priority to kill boss over minions? Sometime that help

Yeah goes for bosses first. Just that infinity cant break his light immune once in a while so merc tries to kill but takes too long and CE gets her.

**Comment by drderpman** [comment]
> Insert the following:
> 
> ```
> var unit = getUnit(1, 526); // Nihlathak
> if (unit && Attack.checkResist(unit, "lightning")) {
> Attack.kill(526); // Nihlathak
> }
> ```

will add this to pindleskin.js script since i go to him via that, let you know if it works

**Comment by odog13** [comment]
> 
> 
> Insert the following:
> 
> ```
> var unit = getUnit(1, 526); // Nihlathak
> if (unit && Attack.checkResist(unit, "lightning")) {
> Attack.kill(526); // Nihlathak
> }
> ```

ok now he is skipping light but how i add light enchanted ?

**Comment by 5noop** [comment]
add the following condition `!unit.getEnchant(17)`

**Comment by odog13** [comment]
> !unit.getEnchant(17)

can you help me set it correct in the nihlatak config ? what i did dont work sorry im not so into code

/**
*	@filename	Nihlathak.js
*	@author		kolton
*	@desc		kill Nihlathak
*/

function Nihlathak() {
	Town.doChores();
	Pather.useWaypoint(123);
	Precast.doPrecast(false);

	if (!Pather.moveToExit(124, true)) {
		throw new Error("Failed to go to Nihlathak");
	}

	Pather.moveToPreset(me.area, 2, 462, 0, 0, false, true);

	if (Config.Nihlathak.ViperQuit && getUnit(1, 597)) {
		print("Tomb Vipers found.");

		return true;
	}

	var unit = getUnit(1, 526); // Nihlathak
	!unit.getEnchant(17)
    if (unit && Attack.checkResist(unit, "lightning"))	{
    Attack.kill(526); // Nihlathak
}
	Pickit.pickItems();

	return true;
}

**Comment by odog13** [comment]
> 
> 
> add the following condition `!unit.getEnchant(17)`

please help me set it right mate copy nihlatak with both lightning and enchant inserted please mate <3

**Comment by 5noop** [comment]
```
if (unit && Attack.checkResist(unit, "lightning") && !unit.getEnchant(17)) {
Attack.kill(526); // Nihlathak
}
```

**Comment by odog13** [comment]
> 
> 
> ```
> if (unit && Attack.checkResist(unit, "lightning") && !unit.getEnchant(17)) {
> Attack.kill(526); // Nihlathak
> }
> ```

this not work bro if i copy paste it at line 24 ... can you post whole nihlatak config so i just have to replace ? sorry for be annoying but it will help me so much ins my runs

**Comment by odog13** [comment]
> 
> 
> ```
> if (unit && Attack.checkResist(unit, "lightning") && !unit.getEnchant(17)) {
> Attack.kill(526); // Nihlathak
> }
> ```

i know that you know how to do it can you please help me bro with fully inserted in nihlatak cfg ? pls <3

**Comment by 5noop** [comment]
```
function Nihlathak() {
	Town.doChores();
	Pather.useWaypoint(123);
	Precast.doPrecast(false);

	if (!Pather.moveToExit(124, true)) {
		throw new Error("Failed to go to Nihlathak");
	}

	Pather.moveToPreset(me.area, 2, 462, 0, 0, false, true);

	if (Config.Nihlathak.ViperQuit && getUnit(1, 597)) {
		print("Tomb Vipers found.");

		return true;
	}

	var unit = getUnit(1, 526); // Nihlathak
	if (unit && Attack.checkResist(unit, "lightning") && !unit.getEnchant(17)) {
		Attack.kill(526); // Nihlathak
	}
	Pickit.pickItems();

	return true;
}
```

**Comment by odog13** [comment]
> 
> 
> ```
> function Nihlathak() {
> 	Town.doChores();
> 	Pather.useWaypoint(123);
> 	Precast.doPrecast(false);
> 
> 	if (!Pather.moveToExit(124, true)) {
> 		throw new Error("Failed to go to Nihlathak");
> 	}
> 
> 	Pather.moveToPreset(me.area, 2, 462, 0, 0, false, true);
> 
> 	if (Config.Nihlathak.ViperQuit && getUnit(1, 597)) {
> 		print("Tomb Vipers found.");
> 
> 		return true;
> 	}
> 
> 	var unit = getUnit(1, 526); // Nihlathak
> 	if (unit && Attack.checkResist(unit, "lightning") && !unit.getEnchant(17)) {
> 		Attack.kill(526); // Nihlathak
> 	}
> 	Pickit.pickItems();
> 
> 	return true;
> }
> ```

very much thanks bro <3
